packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/MarkersController.java

```diff
@@ -210,20 +219,96 @@ public void onAnimationUpdate(ValueAnimator animation) {
       fadeIn.setInterpolator(fadeInInterpolator);
       fadeIn.start();
     }
-    MarkerController controller = new MarkerController(marker, consumeTapEvents);
+    MarkerController controller = new MarkerController(marker, consumeTapEvents, cozyMarkerData);
     markerIdToController.put(markerId, controller);
     googleMapsMarkerIdToDartMarkerId.put(marker.getId(), markerId);
   }
 
-  private void changeMarker(Object marker) {
-    if (marker == null) {
+  private void changeMarker(Object newMarker) {
+    if (newMarker == null) {
       return;
     }
-    String markerId = getMarkerId(marker);
+    String markerId = getMarkerId(newMarker);
     MarkerController markerController = markerIdToController.get(markerId);
     if (markerController != null) {
-      Convert.interpretMarkerOptions(marker, markerController, cozyMarkerBuilder);
+      final CozyMarkerData startCozyMarkerData = markerController.currentCozyMarkerData;
+      final CozyMarkerData endCozyMarkerData = Convert.toCozyMarkerData(newMarker);
+      final boolean isTheSameMarker = Objects.equals(startCozyMarkerData, endCozyMarkerData);
+
+      if(startCozyMarkerData != null && 
+         endCozyMarkerData != null &&
+         endCozyMarkerData.isAnimated && 
+         !isTheSameMarker){
+
+        animateMarkerTransition(markerController, newMarker, startCozyMarkerData, endCozyMarkerData);
+      }else{
+        Convert.interpretMarkerOptions(newMarker, markerController, cozyMarkerBuilder);
+      }
+    }
+  }
+
+  private void animateMarkerTransition(MarkerController markerController, Object newMarker, CozyMarkerData startMarkerData, CozyMarkerData endMarkerData){
+    final List<BitmapDescriptor> bitmapsForFrame = new ArrayList<BitmapDescriptor>();
+    final Deque<Marker> markersQueue = new ArrayDeque<Marker>();
+
+    final int totalNumberOfSimultaneousMarkers = 3;
+    final int expectedFps = 60;
```

Comment:
Should we think about dynamic frames calculation based on device screen properties? 
There's many devices on market now, with 90, 120, 144 hz screens, and 60 hz and will feels like laggy and strobing

Reply:
I didn't find a way to fetch the frame rate dynamically, but if you have, please send me the reference.

About the worst case scenario, I can put a 144, but I'm afraid it will consume too much memory for low-end devices

Reply:
Just extracted this to another class and applied this logic!